src/cdk/v2/destinations/dynamic_yield/procWorkflow.yaml

```diff
@@ -0,0 +1,108 @@
+bindings:
+  - name: EventType
+    path: ../../../../constants
+  - path: ../../../../v0/destinations/dynamic_yield/config
+  - name: removeUndefinedAndNullValues
+    path: ../../../../v0/util
+  - path: ../../bindings/jsontemplate
+
+steps:
+  - name: messageType
+    template: |
+      .message.type.toLowerCase()
+  - name: validateInput
+    template: |
+      let messageType = $.outputs.messageType;
+      $.assert(.message.type, "message Type is not present. Aborting message.");
+      $.assertConfig(.destination.Config.apiKey, "Invalid Api Key");
+      $.assert(messageType in {{$.EventType.([.IDENTIFY, .TRACK])}}, 
+        "message type " + messageType + " is not supported")
+
+  - name: validateInputForTrack
+    description: Additional validation for Track events
+    condition: $.outputs.messageType === {{$.EventType.TRACK}}
+    template: |
+      $.assert(.message.event, "event_name could not be mapped. Aborting.")
+
+  - name: commonFields
+    description: |
+      Builds common fields in destination payload.
+    template: |
+      let commonFields = .message.().({
+        "user": { "id": {{{{$.getGenericPaths("userId")}}}}},
+        "session": {"custom": $.toString({{{{$.getGenericPaths("sessionId")}}}})},
+        "context":{"device":{"ip": .context.ip}}
+      });
+      $.removeUndefinedAndNullValues(commonFields)
+
+  - name: validateCommonFields
+    template: |
+      const userId = $.outputs.commonFields.user.id;
+      const sessionId = $.outputs.commonFields.session.custom;
+      const ip = $.outputs.commonFields.context.device.ip;
+      $.assert(userId, "userId is required");
+      $.assert(sessionId, "sessionId is required");
+      $.assert(ip, "ip is required");
+
+  - name: prepareIdentifyPayload
+    condition: $.outputs.messageType === {{$.EventType.IDENTIFY}}
+    template: |
+
+    steps:
+      - name: propertiesFields
+        template: |
+          let hashedEmail = .message.().(
+            {{{{$.getGenericPaths("email")}}}};
+            );
+          hashedEmail = hashedEmail? sha256(hashedEmail): hashedEmail;
+          let properties = {
+            "hashedEmail": hashedEmail,
+            "dyType": 'identify-v1'
+          };
+          properties = $.removeUndefinedAndNullValues(properties);
+      - name: updatedPropertiesFields
+        condition: $.outputs.prepareIdentifyPayload.propertiesFields.hashedEmail === null
+        template: |
+          let updatedProperties = $.outputs.prepareIdentifyPayload.propertiesFields;
```

Comment:
why do we need this if we are assigning  in line 68

Reply:
Updated